tools/check_format.py

```diff
@@ -70,6 +71,69 @@
     "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
 }
 
+# lookPath searches for the given executable in all directories in PATH
+# environment variable. If it cannot be found, empty string is returned.
+def lookPath(executable):
+  for path_dir in os.environ["PATH"].split(os.pathsep):
+    executable_path = os.path.join(path_dir, executable)
+    if os.path.exists(executable_path):
+      return executable_path
+  return ""
+
+# pathExists checks whether the given path exists. This function assumes that
+# the path is absolute and evaluates environment variables.
+def pathExists(executable):
+  return os.path.exists(os.path.expandvars(executable))
+
+# executableByOthers checks whether the given path has execute permission for
+# others.
+def executableByOthers(executable):
```

Comment:
Why does it matter if the path is executable by others?

Pattern I'm running into involves having tools installed in my home directory, so they are owned and executable by me only.  This check fails when I try to run the script, despite format working fine if this check is skipped.